src/propagate_constant.cpp

```diff
@@ -98,22 +109,31 @@ void propagate_constant::apply(module& m) const
     {
         if(not literals[i].empty())
         {
+            bool skip_pc = false;
             if(enabled(MIGRAPHX_TRACE_PROPAGATE_CONSTANT{}))
-            {
                 std::cout << "Constant replace: " << std::endl;
-                std::vector<instruction_ref> inss;
-                fix([&](auto self, auto ins) {
-                    if(contains(inss, ins))
-                        return;
-                    for(auto input : ins->inputs())
-                        self(input);
-                    inss.push_back(ins);
-                })(const_instrs_vec[i]);
-                m.debug_print(inss);
+            std::vector<instruction_ref> inss;
+            fix([&](auto self, auto ins) {
+                if(contains(inss, ins))
+                    return;
+                if(ins->name() != "@literal" and not is_const_ins(ins, skip_ops))
+                {
+                    skip_pc = true;
+                    return;
+                }
+                for(auto input : ins->inputs())
+                    self(input);
+                inss.push_back(ins);
+            })(const_instrs_vec[i]);
+            if(not skip_pc)
+            {
+                if(enabled(MIGRAPHX_TRACE_PROPAGATE_CONSTANT{}))
+                    m.debug_print(inss);
+
+                assert(literals[i].get_shape() == const_instrs_vec[i]->get_shape());
+                auto l = m.add_literal(literals[i].get_shape(), literals[i].data());
+                m.replace_instruction(const_instrs_vec[i], l);
```

Comment:
Dont refactor propagate_constant, there is already perf regressions across almost all models because its skipping a lot of const folding. 

For now just add dequatizelinear and unpack_int4 to skip_propagate like you did in earlier:

```cpp
bool skip_propagate(instruction_ref ins)
{
    if(ins->name() == "contiguous" or ins->name() == "dequantizelinear")
        return skip_propagate(ins->inputs().front());
    if(ins->name() == "unpack_int4")
        return true;
    auto&& s = ins->get_shape();
    if(s.broadcasted() and not s.scalar())
        return true;
    if(s.scalar() and s.elements() != 1)
        return true;
    return false;
}
```

This will handle the cases with `--int4-weights`. It wont handle other cases from other models, but we can tackle that in a later PR(I dont want this to block the other changes that are being made in this PR) as we need to discuss a better design on handling those cases including the additional cases for block quantization.

Reply:
Thanks. Yes, this simple case was tackled early on. But it was too far removed from what the actual quark-generated model did with its int4 zero_points, which was expected to be supported in the first-deliverable itself.

Reply:
For now, the changes are reverted to handing just `--int4-weights`.